core/src/main/scala/org/apache/spark/SparkContext.scala

```diff
@@ -348,10 +348,27 @@ class SparkContext(config: SparkConf) extends Logging with ExecutorAllocationCli
 
   // Thread Local variable that can be used by users to pass information down the stack
   private val localProperties = new InheritableThreadLocal[Properties] {
-    override protected def childValue(parent: Properties): Properties = new Properties(parent)
```

Comment:
Hi @zsxwing , reasons to use `InheritableThreadLocal` can be seen here (https://github.com/mesos/spark/pull/937). mainly it is used for Spark Streaming with FIFO scheduling strategy.